core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -392,15 +396,22 @@ public IndicesQueriesRegistry buildQueryParserRegistry() {
 
         for (QueryRegistration<?> query : queries) {
             QueryParser<? extends QueryBuilder<?>> parser = query.parser;
-            for (String name: query.names) {
+            for (String name: query.parseField.getAllNamesIncludedDeprecated()) {
```

Comment:
Maybe it's too much for this PR but would it be better for IndicesQueryRegistry to have a `registerQuery(ParserField, QueryParser)` method which then makes sure it builds the parseField and queryParsers maps properly internally?